app/models/concerns/taxonomix.rb

```diff
@@ -132,6 +132,59 @@ def scope_by_taxable_ids(scope)
         scope.where("#{table_name}.id IN (#{cached_ids.join(',')})")
       end
     end
+
+    def join_organizations
+      join_taxonomies(user_organizations, "tto")
+    end
+
+    def join_locations
+      join_taxonomies(user_locations, "ttl")
+    end
+
+    def join_taxonomies(taxonomies, join_name)
+      joins("LEFT JOIN #{TAXONOMY_JOIN_TABLE} AS #{join_name} "\
+        "ON #{table_name}.id = #{join_name}.taxable_id "\
+        "AND #{join_name}.taxable_type = '#{base_class.name}' "\
+        "AND #{join_name}.taxonomy_id IN (#{taxonomies.map(&:id).join(', ')})")
+    end
+
+    def taxonomy_join_scope
+      distinct.join_organizations.join_locations
+    end
+
+    def in_taxonomy_scope(&block)
+      scope = taxonomy_join_scope
+      clauses = []
+
+      unless !Organization.current && User.current.admin?
```

Comment:
@kgaikwad I think you are right. IIRC I want to skip the unnecessary join completely if the user is admin and organization is not set.

Reply:
@kgaikwad I think your are right. IIRC what I want is to skip this clause entirely since the user is an admin and no org is set. Thus, it is not needed to filter anything here.

Reply:
I don't see the purpose of this if branch as well, could you please clarify on this? I would say this is already handled by the `taxonomy_join_scope` :thinking:

Reply:
Let say you have a untaxed audit records which will be "taxonomy.id = null" when you left joins the audits table with taxanomy table. If an admin user doesn't set the organization.current then he/she would like to view the both taxed and untaxed records too. The simple way is to skip the "user_organizations" entirely so that the user can see all the records. Otherwise, we can add another condition like "(taxonomy.id  in (1, 2, 3) or taxonomy_is is null)" which was the way I did in the beginning but later I decided to simplify the query.

Reply:
I think this is what I mentioned in the meeting. The condition here is to match the below:

https://github.com/theforeman/foreman/blob/aae77609c14b1ed2e17d8738e3e18afc256d8417/app/models/concerns/taxonomix.rb#L67

Reply:
@ezr-ondrej